### PR TITLE
chore(cryostat3): various cleanups

### PR DESCRIPTION
**Pull request by andrewazores**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Related to #710

## Description of the change:
- cleans up some unused volume mounts and env vars
- removes Grafana basic auth secret - no longer needed, since Grafana is placed behind the same auth proxy as Cryostat
- configures Grafana for anonymous access (not really anonymous since traffic must go through auth proxy first)
- corrects environment variables so Cryostat UI "View in Grafana" correctly opens Grafana UI
- renames some CR fields to better reflect what they mean in 3.0
- updates docs/config.md
- removes AuthProperties CR field since this is replaced by AuthorizationOptions now

## Motivation for the change:
General clean up, get Grafana working.

## How to manually test:
1. Check out, build, and test PR. Use `OPENSHIFT_OAUTH_PROXY_IMG=quay.io/andrewazores/openshift-oauth-proxy:test-14` (rebuild with golang 1.22) so the UI is usable without the proxy path redirect bug.
2. Create Cryostat CR with cert_manager integration disabled
3. Note Application Route URL, open a new private/incognito browser tab and visit ex. `http://cryostat-cryostat.apps-crc.testing/grafana` to go directly to the Grafana dashboard. This should prompt you to log in with OpenShift, just like visiting the Cryostat UI.
4. In Cryostat UI, log in with OpenShift, then define a `localhost:0` custom target, start a recording, and "View in Grafana" it. This should pass you directly through to Grafana without re-auth.
5. Copy and paste the "View in Grafana" tab URL into a new incognito window to simulate a different, unauthenticated user. Paste and go. You should be prompted to log in.
6. Archive the recording and repeat the same "View in Grafana" tests.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -313,48 +314,39 @@ spec:
...
-data:  
-  auth.properties: |
-    TARGET=pods,deployments.apps
-    RECORDING=pods,pods/exec
-    CERTIFICATE=deployments.apps,pods,cryostats.operator.cryostat.io
-    CREDENTIALS=cryostats.operator.cryostat.io
-```
-
-If custom mapping is specified, a ClusterRole must be defined and should contain permissions for all Kubernetes objects listed in custom permission mapping. This ClusterRole will give additional rules on top of [default rules](https://github.com/cryostatio/cryostat-operator/blob/1b5d1ab97fca925e14b6c2baf2585f5e04426440/config/rbac/oauth_client.yaml).
+On OpenShift, the authentication/authorization proxy deployed in front of the Cryostat application requires all users to pass a `create pods/exec` access review in the Cryostat installation namespace.
```
Comment (tthvo) [inline docs/config.md:321]: Should we mention this is the default and its configurable?

File: docs/config.md [hunk 2]
```diff
@@ -313,48 +314,39 @@ spec:
...
+This means that access to the Cryostat application is granted to exactly the set of OpenShift cluster user accounts and service accounts which have this Role.
 
+The auth proxy may also be configured to allow Basic authentication by creating a Secret containing an `htpasswd` user file. Any user accounts defined in this file will also be granted access to
```
Comment (tthvo) [inline docs/config.md:324]: Would it be helpful to include an example how to generate that secret?

**Comment by ebaron** [comment]
I was able to create a recording from the Topology view, but can't access the Recordings view:
Screenshot 2024-05-07 at 16-56-29 Recordings

**Comment by andrewazores** [comment]
Are you using the plain upstream OpenShift OAuth Proxy or a golang 1.22 rebuild?

**Comment by ebaron** [comment]
> Are you using the plain upstream OpenShift OAuth Proxy or a golang 1.22 rebuild?

It should have been the rebuild. I'll try again, maybe I screwed it up.

**Review by ebaron — APPROVED** [review]
Yeah I didn't override the OAuth image properly. Looks good!

**Pull request merged**